### PR TITLE
add cmake git hooks options

### PR DESCRIPTION
**Pull request by jiayuehua**
#### What type of PR is this?
- [*] bug
- [ ] feature
- [ ] enhancement

#### What does this PR do?

add option that can toggle wether cmake generate git hooks , default ON, So that platform without sh shell can set it OFF and commit code.
#### Checklist：
- [x] Documentation affected （Please add the label if documentation needs to be modified.)
- [ ] Incompatible （If it is incompatible, please describe it and add corresponding label.）
- [ ] Need to cherry-pick （If need to cherry-pick to some branches, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

#### Release notes：
Please confirm whether to reflect in release notes and how to describe:
>                                                                 `

### CODE REVIEW

File: cmake/nebula/GitHooksConfig.cmake [hunk 1]
```diff
@@ -1,19 +1,21 @@
-if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/.git/")
```
Comment (yixinglu) [inline cmake/nebula/GitHooksConfig.cmake:1]:
```cmake
if (ENABLE_CREATE_GIT_HOOKS AND (EXISTS "xxx"))
 # ...
endif()
```

Comment (jiayuehua) [reply cmake/nebula/GitHooksConfig.cmake:1]: fix it.

**Comment by jiayuehua** [comment]
has make if condition more clear.

**Review by yixinglu — APPROVED** [review]
LGTM

**Comment by yixinglu** [comment]
@sherman-the-tank Yuehua uses visual studio IDE to develop nebula project locally. I think it's ok to introduce this option, it will not affect current workflow of others and is convenient for him to develop under windows system. WDYT?

**Comment by jiayuehua** [comment]
Add include what you use to let developer delete no use include files.

File: cmake/IncludeWhatYouUse.cmake [hunk 2]
```diff
@@ -0,0 +1,11 @@
+if(ENABLE_IncludeWhatYouUse)
+  find_program(IncludeWhatYouUse include-what-you-use)
```
Comment (yixinglu) [inline cmake/IncludeWhatYouUse.cmake:2]: `IncludeWhatYouUse_FOUND`

File: CMakeLists.txt [hunk 3]
```diff
@@ -26,6 +26,8 @@ project("Nebula Graph" C CXX)
 option(ENABLE_PACK_ONE "Whether to package into one" ON)
 option(ENABLE_VERBOSE_BISON "Enable Bison to report state" OFF)
 option(ENABLE_PACKAGE_TAR "Enable package artifacts to tar." OFF)
+option(ENABLE_CREATE_GIT_HOOKS "Enable create git hooks." ON)
+option(ENABLE_IncludeWhatYouUse "Enable Include_What_You_Use" OFF)
```
Comment (sherman-the-tank) [inline CMakeLists.txt:30]: Please use all CAPITAL, like other options, following the convention

File: cmake/IncludeWhatYouUse.cmake [hunk 4]
```diff
@@ -0,0 +1,11 @@
+if(ENABLE_IncludeWhatYouUse)
```
Comment (sherman-the-tank) [inline cmake/IncludeWhatYouUse.cmake:1]: Suggest to place this file in cmake/nebula

File: cmake/IncludeWhatYouUse.cmake (hunk 2 above)
Comment (yixinglu) [reply cmake/IncludeWhatYouUse.cmake:2]: Why not to turn on iwyu when find the program automatically? i think this option is not necessary.

**Review by sherman-the-tank — COMMENTED** [review]
Well done!

Just couple styling comments

File: cmake/IncludeWhatYouUse.cmake (hunk 4 above)
Comment (jiayuehua) [reply cmake/IncludeWhatYouUse.cmake:1]: should place in cmake. This is a general cmake tool

Comment (jiayuehua) [reply cmake/IncludeWhatYouUse.cmake:1]: ok I change it to CAPITAL

File: cmake/IncludeWhatYouUse.cmake (hunk 2 above)
Comment (jiayuehua) [reply cmake/IncludeWhatYouUse.cmake:2]: because it run very slow,  Just let developer run it  .

**Review by yixinglu — APPROVED** [review]

**Review by Aiee — APPROVED** [review]

**Pull request merged**